src/OpenSage.Game/Radar.cs

```diff
@@ -1,94 +1,40 @@
-﻿using System;
+﻿#nullable enable
+
+using System.Collections.Frozen;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.IO;
 using System.Numerics;
-using OpenSage.Content;
-using OpenSage.Gui;
 using OpenSage.Logic.Object;
 using OpenSage.Mathematics;
-using Veldrid;
 
 namespace OpenSage
 {
     public sealed class Radar : IPersistableObject
     {
-        private readonly Scene3D _scene;
-        private readonly Texture _miniMapTexture;
+        public RadarItemCollection VisibleItems => _visibleItems;
+        public RadarItemCollection HiddenItems => _hiddenItems;
 
-        private readonly RadarItemCollection _visibleItems;
-        private readonly RadarItemCollection _hiddenItems;
+        // fixed-length array
+        public IReadOnlyList<RadarEvent> RadarEvents => _radarEvents;
```

Comment:
I think ReadOnlySpan would be more optimal here, as long as the call sites can work with that. ReadOnlySpan allows for iteration without memory allocation.

Reply:
Great catch!

Reply:
Updated, let me know if that's what you had in mind. From what I can tell it has to be new'd up since it can't be stored as a field unless it's a ref struct (I mean you don't _have_ to explicitly new it up, but if you don't it will via an implicit cast so I figured best to make it explicit)

Reply:
That’s what I meant, nice. I’m fine with the implicit cast though, I think it’s a common-enough pattern for arrays / ReadOnlySpan that it doesn’t need to be explicitly called out.

Reply:
Updated to an implicit cast. I'll trust your judgement of what's more common in this area :)